exporters/committime/collector_image.py

```diff
@@ -0,0 +1,205 @@
+
+        # First get metrics within Labels
+        attribute_mapping = (
+            ImageCommitCollector._DOCKER_LABEL_MAPPING.items() if image_labels else []
+        )
+        for attr_name, label in attribute_mapping:
+            value = image_labels.get(label)
+            if value:
+                if attr_name == "commit_time":
+                    logging.debug(
+                        "Commit time for image %s provided by '%s' label: %s",
+                        metric.image_hash,
+                        label,
+                        value,
+                    )
+                setattr(metric, attr_name, value)
+
+        if not metric.commit_time:
+            metric = self._set_commit_time_from_annotations(metric, errors)
+
+        if not metric.commit_hash:
+            # We ignore all the errors by passing [], because commit hash isn't required.
+            metric = self._set_commit_hash_from_annotations(metric, [])
+
+        metric = self._set_commit_timestamp(metric, errors)
+
+        if not metric.namespace:
+            metric.namespace = "None"
+
+        return metric
+
+    def _set_commit_timestamp(self, metric, errors) -> CommitMetric:
+        # Only convert when commit_time is in metric, previously should be
+        # found from the Label with fallback to annotation
+        if metric.commit_time:
+            metric.commit_timestamp = pelorus.convert_date_time_to_timestamp(
+                metric.commit_time, self._timedate_format
+            )
+        return metric
+
+    def get_commit_time(self, metric) -> Optional[CommitMetric]:
+        return super().get_commit_time(metric)
+
+    def _set_commit_time_from_annotations(
+        self, metric: CommitMetric, errors: list
+    ) -> CommitMetric:
+        if not metric.commit_time:
+            commit_time_annotation = get_env_var(
+                COMMIT_DATE_ANNOTATION_ENV,
+                CommitMetric._ANNOTATION_MAPPIG.get("commit_time"),
+            )
+            commit_time = metric.annotations.get(commit_time_annotation)
+            if commit_time:
+                metric.commit_time = commit_time
+                logging.debug(
+                    "Commit time for image %s provided by '%s' annotation: %s",
+                    metric.image_hash,
+                    commit_time_annotation,
+                    metric.commit_time,
+                )
+            else:
+                errors.append(
+                    "Couldn't get commit time from annotations nor image label"
+                )
+        return metric
+
+    # overrides collector_base.generate_metric()
+    def generate_metrics(self) -> Iterable[CommitMetric]:
+
+        # Initialize metrics list
+        metrics = []
+        app_label = pelorus.get_app_label()
+
+        logging.debug("Searching for images with label: %s" % app_label)
+
+        v1_images = self._kube_client.resources.get(
+            api_version="image.openshift.io/v1", kind="Image"
```

Comment:
So.. FYI.. both the pelorus and basic-python-tekton images are not showing up w/ kind=Image, they are showing up w/

```
 v1_images = self._kube_client.resources.get(
            api_version="image.openshift.io/v1", kind="ImageStreamTag"
```
Perhaps we can make calls to both Image && ImageStreamTag and combine the results in the list.

still testing :) but this should work w/ just minor changes to the tekton pipeline

Reply:
I'm seeing this behavior too. I just noticed this section on the docs for the `Image` resource:

> Image is an immutable representation of a container image and metadata at a point in time. Images are named by taking a hash of their contents (metadata and content) and any change in format, content, or metadata results in a new name. **The images resource is primarily for use by cluster administrators and integrations like the cluster image registry - end users instead access images via the imagestreamtags or imagestreamimages resources.** While image metadata is stored in the API, any integration that implements the container image registry API must provide its own storage for the raw manifest data, image config, and layer contents.

(Emphasis mine).

I didn't notice this before, darn. Are there any images that an ImageStreamTag _won't_ catch?

Reply:
I don't think he ended up searching for imagestreams, just images.
https://github.com/konveyor/pelorus/pull/605/files#diff-916d4878c75e654078b2a2026ea6958b7a222857c5968b4181b517fa3dd6bf07R164

Reply:
No it's currently based on the Image, if there is need for imagestream I would do this as another PR so we don't circle around here.